packages/drivers/routerlicious-driver/src/deltaStorageService.ts

```diff
@@ -69,18 +69,19 @@ export class DeltaStorageService implements IDeltaStorageService {
             "x-correlation-id": uuid.v4(),
         };
 
-        const storageToken = await this.tokenProvider.fetchStorageToken(
-            tenantId,
-            id,
+        const getStorageServiceAuthHeader = async () => {
+            const storageToken = await this.tokenProvider.fetchStorageToken(tenantId, id);
+            return `Basic ${storageToken.jwt}`;
+        };
+        const ops = await authorizedRequestWithRetry<ISequencedDocumentMessage[]>(
+            {
+                method: "get",
+                url: `${this.url}?${query}`,
+                headers,
+            },
+            getStorageServiceAuthHeader,
         );
 
-        if (storageToken) {
-            headers.Authorization = `Basic ${storageToken.jwt}`;
-        }
-
-        const ops = await Axios.get<ISequencedDocumentMessage[]>(
-            `${this.url}?${query}`, { headers });
-
         if (this.logger) {
```

Comment:
Unrelated to your change, but I'd suggest making logger required through all layers. One can always use Null logger in test / non-prod environments